File: node/zkgroup/profiles/ProfileKeyCredentialRequest.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class ProfileKeyCredentialRequest extends ByteArray {

  static SIZE = 329;

  constructor(contents: Buffer) {
    super(contents, ProfileKeyCredentialRequest.SIZE, true);
    NativeImpl.ProfileKeyCredentialRequest_CheckValidContents(contents);
  }

}

File: java/shared/java/org/signal/libsignal/zkgroup/profiles/ProfileKeyCredentialRequest.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.profiles;

import static org.signal.libsignal.internal.FilterExceptions.filterExceptions;

import org.signal.libsignal.internal.Native;
import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class ProfileKeyCredentialRequest extends ByteArray {
  public ProfileKeyCredentialRequest(byte[] contents) throws InvalidInputException {
    super(contents);
    filterExceptions(
        InvalidInputException.class,
        () -> Native.ProfileKeyCredentialRequest_CheckValidContents(contents));
  }
}
